climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+        project = partial(
+            pyproj.transform,
+            pyproj.Proj(init=DEF_CRS['init']), # source coordinate system
+            pyproj.Proj(init='epsg:3310')) # destination coordinate system: albers california
+
+        concave_hull_m = transform(project, concave_hull)  # apply projection
+        area_hull_one_fire = concave_hull_m.area/10000
+
+        # Plot the polygone around the fire
+        if return_plot:
+            u_alpha.plot_polygon(concave_hull)
+            plt.plot(fire_lon, fire_lat, 'o', color='red', markersize=0.5)
+
+        return area_hull_one_fire
+
+    def set_hist_fire_seasons_FIRMS(self, csv_firms, centr_res_factor=1,
+                                    centroids=None, hemisphere=None,
+                                    year_start=None, year_end=None,
+                                    keep_all_fires=False):
+
+        """ Parse FIRMS data and generate historical fire seasons.
+        
+        Individual fires are created using temporal and spatial clustering
+        according to the 'set_hist_fire' method. fires are then summarized
+        using max intensity at each centroid for each year.
+
+        Parameters
+        ----------
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+                (https://firms.modaps.eosdis.nasa.gov/download/)
+            centr_res_factor : int, optional, default=1
+                resolution factor with respect to the satellite data to use
+                for centroids creation
+            centroids : Centroids, optional
+                centroids in degrees to map data
+            hemisphere : str, optional
+                'SHS' or 'NHS' to define fire seasons
+            year_start : int, optional
+                start year; FIRMS fires before that are cut; no cut if not
+                specified
+            year_end : int, optional
+                end year; FIRMS fires after that are cut; no cut if not
+                specified
+            keep_all_fires : bool, optional
+                keep detailed list of all fires; default is False to save
+                memory.
+
+        Returns
+        -------
+            self : climada.hazard.WildFire instance
+
+        """
+
+        LOGGER.info('Setting up historical fires for year set.')
+        self.clear()
+
+        # read and initialize data
+        firms = self._clean_firms_csv(csv_firms)
+        # compute centroids
+        res_data = self._firms_resolution(firms)
+        if not centroids:
+            centroids = self._firms_centroids_creation(firms, res_data, centr_res_factor)
+        else:
+            if not centroids.coord.size:
+                centroids.set_meta_to_lat_lon()
+
+        # define hemisphere
+        if hemisphere is None:
+            if centroids.lat[0] > 0:
+                hemisphere = 'NHS'
+            elif centroids.lat[0] < 0:
+                hemisphere = 'SHS'
```

Comment:
Why would the first centroid latitude define whether it is North of South hemisphere? It could be the whole Earth for instance.

Reply:
I've adjusted the docsting. I think it is good enough to just test for the first centroid, if nothing is specified. For countries/regions across the equator it can and should be specified manually

Reply:
This is not clear in the docstrings. How is a fire season defined?

Also, you cannot under the hood assume a hemisphere and not handle obvious problematic cases. This must be adjusted. When centroids cross the equator, an error must be raised as it would be an error!

Reply:
To be fair I don't see that many problematic cases arise. For countries which are on the equator, the seasonality doesn't matter anyway (as they're in the tropics and thus don't have very strong seasons).

I've adjusted the code so that it raises a warning if not all centroids are on the same hemisphere! See commit 4d3bb47c

Reply:
But if the seasons are not very strong, how are the wildfire in one year bunched together?

Reply:
well seasonality in the tropics is not very strong. on higher/lower latitudes that becomes more important =) I think in the tropics it's fair to just make a cut "somewhen"